src/Doctrine/Orm/Extension/PaginationExtension.php

```diff
@@ -80,6 +80,10 @@ public function getResult(QueryBuilder $queryBuilder, string $resourceClass = nu
             $query->setHint(CountWalker::HINT_DISTINCT, false);
         }
 
+        if (null !== $distinctCount = $operation?->getDistinctCount()) {
```

Comment:
This should probably be a `stateOption` instead (see the Options object on the ORM namespace). I'd keep the option but we should probably check that the JOIN doesn't exist.